docs/duration-calculator.md

```diff
@@ -0,0 +1,68 @@
+# Duration Calculator
+
+The number of beats in a measure depend on the time signature. The top number 
+of the time signature determines the number of beats per measure, while the 
+bottom one determines the length of one beat. 
+
+For example:
+
+4/4: 4 beats of 1/4 notes.
+6/8: 6 beats of 1/8 notes.
+
+## Note Measurement
+
+### Convert Note Duration Into TPQN
+
+Calculate measure budget based on time signature. Ideally, it is not ideal to
+work with fractions, so it would be better to express lengths w.r.t the shorest
+note.
+
+https://en.wikipedia.org/wiki/Two_hundred_fifty-sixth_note
+
+The shortest recorded note is 1/256, which means that we would have to
+use at least 1/512 to account for a 1/256 dot length, or 1/1024 to account for a
+double dot. 
+
+Lets say we go with 1024: 
+
+
+Note Duration, Note Weight
+1 = 1024
+1/2 = 512
+1/4 = 256
+1/8 = 128
+1/16 = 64
+1/32 = 32
+1/64 = 16
+1/128 = 8
+1/256 = 4
+1/512 = 2
+1/1024 = 1 for a note: (1/256)+dot+dot
+
+Another solution is to make the calculations w.r.t to the cuarter note (1/4).
+This is nice because it would make us more compatible with MIDI (see 
+https://en.wikipedia.org/wiki/Pulses_per_quarter_note). With this in mind, 
+we can probably use Ticks Per Quarter Note (TPQN).
+
+Music XML supports 1/1024 notes:
+http://usermanuals.musicxml.com/MusicXML/Content/ST-MusicXML-note-type-value.htm
+
+### Implementation
+
+- Make TPQN configurable (e.g 1024, TODO: look into MIDI time duration). This
+  parameter is not necessarily exposed in the public API.
+- Calculate budget in terms of TPQN. 4/4 = 4 * TPQN, 6 / 8  = 6 * TPQN / 2
+
+## Measure Calculations
+
+- Measure occupancy value (measure usage): used space / measure budget. 
+- Insert note. Suggest available note durations given used space. 
+- Delete note.
+- Replace note. This can be a combination of delete followed by insert?
+
+Opens:
+
+- Good to have the ability to tell what can be used to fill the measure.
+- Let the API caller know that the measure is incomplete.
```

Comment:
My thinking was we will only perform validation upon the specific validation method being called so that the caller can decide if they want to be able to edit without validating until a certain point or be validating after every modification. Maybe we can also have a mode that can be turned on to automatically validate after every one or something like that. However, I don't think that's necessary for the first pass.